src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            table = resize();
+        }
+        int index = getIndex(key, table);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        Node<K, V> nextNode = table[index];
+        while (nextNode.next != null) {
+            if (Objects.equals(nextNode.key, key)) {
+                nextNode.value = value;
+                return;
+            }
+            nextNode = nextNode.next;
+        }
+        if (Objects.equals(nextNode.key, key)) {
+            nextNode.value = value;
+            return;
+        }
+        nextNode.next = new Node<>(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
+        int index = getIndex(key, table);
+        Node<K, V> nextPair = table[index];
+        while (nextPair != null) {
+            if (Objects.equals(nextPair.key, key)) {
+                value = nextPair.value;
```

Comment:
Do we need to continue the loop if find the key?
```suggestion
                return nextPair.value;
```